### PR TITLE
Fixes Fire Pillar item requirement check

### LINKED ISSUE #5266: Fire pillar Skill can't check Blue Gemstone
* **rAthena Hash**: latest

* **Client Date**: exe 2018-06-20

* **Server Mode**: pre-re

* **Description of Issue**: Fire pillar Skill can't check Blue Gemstone
- Would like to know and suggestions or current information about Fire Pillar

1. Why is there no condition in using Blue Gemstone?

2. There is no condition in the Skill_db.yml file for using BlueGemstone.

  * Result: 

Picture 1: The character Gravitation Field does not have Blue Gemstone. When used, the effect does not work and calls for 1 Blue Gemstone.
image

Picture 2: Safety Wall skills in the character do not have Blue Gemstone. When used, the effect does not work and calls for 1 Blue Gemstone.
image

Picture 3: Fire Pillar's character does not have Blue Gemstone and is tested using the skill level 10. The skill level of 6 ~ 10 is to have 1 Blue Gemstone, but ... after using the skill Unchecked Blue Gemstone. The skill works normally.
image

 
  * Expected Result: Fire Pillar LV6-10 can check the Blue Gemstone. 
  * How to Reproduce: Fire Pillar in the character does not have Blue Gemstone and test using the skill level 10. The skill level 6 ~ 10 is to have 1 Blue Gemstone, but ... When the skill is used and not checked. Blue skill works normally
  * Official Information:Skill Fire Pillar LV6-10 must use Blue Gemstone 1 ea.

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #5266

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Fire Pillar requires a Blue Gemstone for levels 6-10.
Thanks to @Pringle012!

### CODE REVIEW

File: src/map/skill.cpp [hunk 1]
```diff
@@ -16937,9 +16937,11 @@ struct s_skill_condition skill_get_requirement(struct map_session_data* sd, uint
 		case SO_FIRE_INSIGNIA:
 		case SO_WIND_INSIGNIA:
 		case SO_EARTH_INSIGNIA:
-		case WZ_FIREPILLAR: // no gems required at level 1-5 [celest]
 			req.itemid[0] = skill->require.itemid[min(skill_lv-1,MAX_SKILL_ITEM_REQUIRE-1)];
```
Comment (lighta) [inline src/map/skill.cpp:16940]: judging from this, we could set different itemid per lvl, did we break something or ?
can't our skilldb.yaml support this ?

File: src/map/skill.cpp [hunk 2]
```diff
@@ -16937,9 +16937,11 @@ struct s_skill_condition skill_get_requirement(struct map_session_data* sd, uint
...
 			req.amount[0] = skill->require.amount[min(skill_lv-1,MAX_SKILL_ITEM_REQUIRE-1)];
+		case WZ_FIREPILLAR: // no gems required at level 1-5 [celest]
+			if (skill_id == WZ_FIREPILLAR && skill_lv < 6)
+				break;
 			level_dependent = true;
```
Comment (gen1x8) [inline src/map/skill.cpp:16945]: is it to comply with skill_db.yml, level_dependent should be false for item requirement ?

```suggestion
		case WZ_FIREPILLAR: // no gems required at level 1-5 [celest]
			if (skill_id == WZ_FIREPILLAR && skill_lv > 5)
				level_dependent = false;
			else
				level_dependent = true;
```

File: doc/skill_db.txt [hunk 3]
```diff
@@ -729,6 +729,29 @@ Sequence Map Form
 
 ItemCost: Item required to cast.
 
+Levels 1 - 5 have no item cost but levels 6 - 10 require a Blue Gemstone.
+  ItemCost:
+    - Item: Blue_Gemstone
+      Amount: 1
+	  Level: 6
```
Comment (Lemongrass3110) [inline doc/skill_db.txt:736]: Dangerous tabs around here!

File: src/map/skill.cpp [hunk 4]
```diff
@@ -16937,7 +16937,6 @@ struct s_skill_condition skill_get_requirement(struct map_session_data* sd, uint
 		case SO_FIRE_INSIGNIA:
 		case SO_WIND_INSIGNIA:
 		case SO_EARTH_INSIGNIA:
-		case WZ_FIREPILLAR: // no gems required at level 1-5 [celest]
 			req.itemid[0] = skill->require.itemid[min(skill_lv-1,MAX_SKILL_ITEM_REQUIRE-1)];
 			req.amount[0] = skill->require.amount[min(skill_lv-1,MAX_SKILL_ITEM_REQUIRE-1)];
 			level_dependent = true;
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:16942]: If you already add support for this into YAML now:
Shouldn't all of these skills be migrated to the new format?

Also please do not forget to increase the version in the header! And I think you already mentioned yourself that the documentation needs to be changed inside the YAML files.

File: src/map/skill.hpp [hunk 5]
```diff
@@ -299,7 +300,7 @@ struct s_skill_db {
 
 class SkillDatabase : public TypesafeCachedYamlDatabase <uint16, s_skill_db> {
 public:
-	SkillDatabase() : TypesafeCachedYamlDatabase("SKILL_DB", 1) {
+	SkillDatabase() : TypesafeCachedYamlDatabase("SKILL_DB", 2) {
```
Comment (Lemongrass3110) [inline src/map/skill.hpp:303]:
```suggestion
	SkillDatabase() : TypesafeCachedYamlDatabase("SKILL_DB", 2, 1) {
```

File: src/tool/csv2yaml.cpp [hunk 6]
```diff
@@ -2233,6 +2233,27 @@ static bool skill_parse_row_skilldb(char* split[], int columns, int current) {
 
 					body << YAML::Key << "Item" << YAML::Value << *item_name;
 					body << YAML::Key << "Amount" << YAML::Value << it_req->second.amount[i];
+
+					uint16 cost_level = i;
+
+					switch (skill_id) { // List of level dependent item costs
+						case WZ_FIREPILLAR:
+							cost_level += 5; // Levels 1-5 have no cost
```
Comment (Lemongrass3110) [inline src/tool/csv2yaml.cpp:2241]:
```suggestion
							if( cost_level < 6) break; // Levels 1-5 have no cost
```

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**